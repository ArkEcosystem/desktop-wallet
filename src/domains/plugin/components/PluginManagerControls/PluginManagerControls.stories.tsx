import React from "react";

import { PluginManagerControls } from "./PluginManagerControls";

export default { title: "Plugins / components / Plugin Manager Controls" };

export const Default = () => {
	const [viewType, setViewType] = React.useState("grid");

	return (
		<div className="flex justify-end">
			<PluginManagerControls
				selectedViewType={viewType}
				onSelectListView={() => setViewType("list")}
				onSelectGridView={() => setViewType("grid")}
			/>
		</div>
	);
};
