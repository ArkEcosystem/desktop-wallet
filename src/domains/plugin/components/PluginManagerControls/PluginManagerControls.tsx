
import { Divider } from "app/components/Divider";
import { Dropdown } from "app/components/Dropdown";
import { LayoutControls } from "app/components/LayoutControls";
import React from "react";

type PluginManagerControlsProps = {
	onSelectGridView: any;
	onSelectListView: any;
	selectedViewType: string;
};

export const PluginManagerControls = ({
	onSelectGridView,
	onSelectListView,
	selectedViewType,
}: PluginManagerControlsProps) => {
	return (
		<div data-testid="PluginManagerControls" className="flex items-center mx-4">
			<LayoutControls
				onSelectGridView={onSelectGridView}
				onSelectListView={onSelectListView}
				selectedViewType={selectedViewType}
			/>

			<Divider type="vertical" />

			<div className="relative inline-block text-theme-primary-400">
				<Dropdown toggleIcon="Filters">
					<div className="py-8 w-128 px-11" />
				</Dropdown>
			</div>
		</div>
	);
};

PluginManagerControls.defaultProps = {
	selectedViewType: "grid",
};
