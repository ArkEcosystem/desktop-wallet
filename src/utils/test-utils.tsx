import { render } from "@testing-library/react";
import { i18n } from "app/i18n";
import React from "react";
import { I18nextProvider } from "react-i18next";

const WithProviders = ({ children }: { children: React.ReactNode }) => {
	return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};

const customRender = (component: React.ReactElement, options: any) =>
	render(component, { wrapper: WithProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
