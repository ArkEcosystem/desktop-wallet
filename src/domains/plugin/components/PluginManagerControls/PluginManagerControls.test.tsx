import { act, renderHook } from "@testing-library/react-hooks";
import React from "react";
import { fireEvent, render } from "test-utils";

import { PluginManagerControls } from "./PluginManagerControls";

describe("PluginManagerControls", () => {
	it("should render", () => {
		const { asFragment, getByTestId } = render(<PluginManagerControls />);

		expect(getByTestId("PluginManagerControls")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should toggle between list and grid", () => {
		const { result } = renderHook(() => {
			const [viewType, setViewType] = React.useState("grid");

			return {
				viewType,
				setViewType,
			};
		});

		const { asFragment, getByTestId, rerender } = render(
			<PluginManagerControls
				selectedViewType={result.current.viewType}
				onSelectListView={() => result.current.setViewType("list")}
				onSelectGridView={() => result.current.setViewType("grid")}
			/>,
		);

		const gridIcon = getByTestId("LayoutControls__grid--icon");
		const listIcon = getByTestId("LayoutControls__list--icon");

		expect(gridIcon).toHaveClass("text-theme-danger-300");
		expect(listIcon).not.toHaveClass("text-theme-danger-300");

		act(() => {
			fireEvent.click(listIcon);
		});

		rerender(
			<PluginManagerControls
				selectedViewType={result.current.viewType}
				onSelectListView={() => result.current.setViewType("list")}
				onSelectGridView={() => result.current.setViewType("grid")}
			/>,
		);

		expect(gridIcon).not.toHaveClass("text-theme-danger-300");
		expect(listIcon).toHaveClass("text-theme-danger-300");

		act(() => {
			fireEvent.click(gridIcon);
		});

		rerender(
			<PluginManagerControls
				selectedViewType={result.current.viewType}
				onSelectListView={() => result.current.setViewType("list")}
				onSelectGridView={() => result.current.setViewType("grid")}
			/>,
		);

		expect(gridIcon).toHaveClass("text-theme-danger-300");
		expect(listIcon).not.toHaveClass("text-theme-danger-300");

		expect(asFragment()).toMatchSnapshot();
	});
});
