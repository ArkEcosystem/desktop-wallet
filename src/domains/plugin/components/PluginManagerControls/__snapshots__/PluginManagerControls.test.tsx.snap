// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginManagerControls should render 1`] = `
<DocumentFragment>
  <div
    class="flex items-center"
    data-testid="PluginManagerControls"
  >
    <div
      class="flex items-center"
    >
      <div
        class="inline-block"
        data-testid="LayoutControls__grid"
      >
        <div
          class="px-3 py-3 h-full border-b-3 cursor-pointer text-theme-danger-300 border-theme-danger-contrast"
          data-testid="LayoutControls__grid--icon"
        >
          <div
            class="sc-AxjAm jEKiJj"
            height="14"
            width="28"
          >
            <svg>
              grid.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="inline-block"
        data-testid="LayoutControls__list"
      >
        <div
          class="px-3 py-3 h-full border-b-3 cursor-pointer text-theme-primary-400 border-transparent"
          data-testid="LayoutControls__list--icon"
        >
          <div
            class="sc-AxjAm eFgodT"
            height="14"
            width="24"
          >
            <svg>
              list.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PluginManagerControls should toggle between list and grid 1`] = `
<DocumentFragment>
  <div
    class="flex items-center"
    data-testid="PluginManagerControls"
  >
    <div
      class="flex items-center"
    >
      <div
        class="inline-block"
        data-testid="LayoutControls__grid"
      >
        <div
          class="px-3 py-3 h-full border-b-3 cursor-pointer text-theme-danger-300 border-theme-danger-contrast"
          data-testid="LayoutControls__grid--icon"
        >
          <div
            class="sc-AxjAm jEKiJj"
            height="14"
            width="28"
          >
            <svg>
              grid.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="inline-block"
        data-testid="LayoutControls__list"
      >
        <div
          class="px-3 py-3 h-full border-b-3 cursor-pointer text-theme-primary-400 border-transparent"
          data-testid="LayoutControls__list--icon"
        >
          <div
            class="sc-AxjAm eFgodT"
            height="14"
            width="24"
          >
            <svg>
              list.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
